Lesson16/selcan/script.js

```diff
@@ -0,0 +1,304 @@
+function offerJob(salary,experience){
+	if (salary<30.000 && experience<5){
+		return "Sorry, we cannot offer you the job at this time";
+	} else if(salary<50.000 && experience>5 && experience<10){
+		return "We can offer you the job at a lower salary";
+	} else if(salary>=50.000 && experience>=10){
+		return "We can offer you the job at the requested salary";
+	}
+}
+console.log(offerJob(salary,experience));
+
+
+// 12. Write a function that asks the user to enter their score on a test. If the score is less than 60, print "You failed the test". If the score is between 60 and 70 (inclusive), print "You passed the test, but your grade is not great". If the score is between 71 and 80 (inclusive), print "You got a B". If the score is between 81 and 90 (inclusive), print "You got an A". If the score is greater than 90, print "You got an A+ - Great job!".
+function score(number){
+	if (number<60){
+		return "You failed the test";
+	} else if(number>=60 && number<=70){
+		return "You passed the test, but your grade is not great";
+	} else if(number>71 && number<=80){
+		return "You got a B";
+	} else if(number>81 && number<=90){
+		return "You got an A";
+	} else if(number>90){
+		return "You got an A+ - Great job!";
+	}
+}
+console.log(score(85));
+
+// 13. Write a function that asks the user to enter a month number (1-12) and displays the corresponding season name. Use a switch statement to handle the different cases.
+
+function getSeasonName(numberOfAMonth) {
+	switch (numberOfAMonth) {
+      // numberOfAMonth === 12
+		case 12:
+		case 1:
+		case 2:
+      // console.log('Winter');
+			// break;
+			return 'Winter';
+		case 3:
+		case 4:
+		case 5:
+			return 'Spring';
+		case 6:
+		case 7:
+		case 8:
+			return 'Summer';
+		case 9:
+		case 10:
+		case 11:
+			return 'Autumn';
+    default: 
+      return 'Value is not supported, please enter value between 1 and 12.';
+	}
+}
+
+console.log('Exercise 13', getSeasonName(1), getSeasonName(5), getSeasonName(20));
+
+// 14. Write a function that asks the user to enter a grade letter (A, B, C, D, or F) and displays a message indicating whether the grade is passing (A, B, C) or failing (D, F). Use a switch statement to handle the different cases.
+function gradeLetter(letter){
+	letter=letter.toUpperCase();
+	switch(letter){
+		case A:
+		case B:
+		case C:
+			return "you passed";
+		case D:
+		case F:
+			return "you failed";
+		default: 
+            return "Value is not supported";
+	}
+}
+console.log(gradeLetter(D));
+// 15. Write a function that asks the user to enter their age, and then calculates and displays their age in dog years (where one human year is equivalent to seven dog years). If the user enters a negative number or zero, display an error message.
+function isYear(age){
+	let dogYear=age * 7;
```

Comment:
When you have a case that is not supported in the function, add it before doing any calculation of the result so that you don't run code unnecessary. 

```
function isYear(age){
  if (age <= 0) {
    return "Value is not supported"
  } else {
    return age * 7;
  }
}

```